src/subdomains/staking/application/services/staking-reward.service.ts

```diff
@@ -36,13 +37,32 @@ export class StakingRewardService {
     private readonly outService: StakingRewardOutService,
     private readonly assetService: AssetService,
     private readonly stakingService: StakingService,
+    private readonly rewardRouteRepo: RewardRouteRepository,
+    private readonly stakingRepo: StakingRepository,
   ) {}
 
   //*** PUBLIC API ***//
 
-  async createReward(stakingId: number, dto: CreateRewardDto): Promise<void> {
-    const staking = await this.repository.findOneBy({ id: stakingId });
-    if (!staking) throw new NotFoundException('Staking not found');
+  async createReward(dto: CreateRewardDto): Promise<void> {
+    const rewardRoute = await this.rewardRouteRepo.findOne({
+      where: { id: dto.rewardRouteId },
+      relations: ['strategy'],
+    });
+    const stakings = await this.stakingRepo.findBy({ userId: rewardRoute.strategy.userId });
```

Comment:
Can we load the stakings directly using `relations: ['strategy', 'strategy.stakings']`?